Kudu.Services/ServiceHookHandlers/FetchHandler.cs

```diff
@@ -97,27 +97,35 @@ public class FetchHandler : HttpTaskAsyncHandler
                         context.Response.StatusCode = (int)HttpStatusCode.Accepted;
                         context.ApplicationInstance.CompleteRequest();
                         return;
-                    case FetchDeploymentRequestResult.ForbiddenScmDisabled:
+                    case FetchDeploymentRequestResultStatus.ForbiddenScmDisabled:
                         context.Response.StatusCode = (int)HttpStatusCode.Forbidden;
                         context.ApplicationInstance.CompleteRequest();
                         _tracer.Trace("Scm is not enabled, reject all requests.");
                         return;
-                    case FetchDeploymentRequestResult.ConflictAutoSwapOngoing:
+                    case FetchDeploymentRequestResultStatus.ConflictAutoSwapOngoing:
                         context.Response.StatusCode = (int)HttpStatusCode.Conflict;
                         context.Response.Write(Resources.Error_AutoSwapDeploymentOngoing);
                         context.ApplicationInstance.CompleteRequest();
                         return;
-                    case FetchDeploymentRequestResult.Pending:
+                    case FetchDeploymentRequestResultStatus.Pending:
                         // Return a http 202: the request has been accepted for processing, but the processing has not been completed.
                         context.Response.StatusCode = (int)HttpStatusCode.Accepted;
                         context.ApplicationInstance.CompleteRequest();
                         return;
-                    case FetchDeploymentRequestResult.ConflictDeploymentInProgress:
+                    case FetchDeploymentRequestResultStatus.ConflictDeploymentInProgress:
                         context.Response.StatusCode = (int)HttpStatusCode.Conflict;
                         context.Response.Write(Resources.Error_DeploymentInProgress);
                         context.ApplicationInstance.CompleteRequest();
                         break;
-                    case FetchDeploymentRequestResult.RanSynchronously:
+                    case FetchDeploymentRequestResultStatus.RanSynchronously:
```

Comment:
This new behavior is good for PushDeploymentController since it's new and likely to be used interactively, but here for FetchHandler, please put this bit back how it was (have RanSynchronously fall through to the default case), and have RanSynchronouslyFailed fall through to the default case as well. I'm hesitant to change the behavior here, since this is generally called by webhooks from many different CI services and I don't want to break expectations related to how they react to the response. A short comment to acknowledge the RanSynchronouslyFailed case and explaining that we're avoiding a behavior change would be good also.

Reply:
Reverted to default case